### DISCUSSION TITLE
Feature request: Ability to center/align

### DISCUSSION THREAD

**Discussion by biggestsonicfan** [dune3d/dune3d#53] [Q&A] [answered]
I've imported my own .step file and adjusted a workspace area, however when drawing a reference rectangle there appears to be zero way to align the rectangle to the center of the workspace.

![Screenshot 2024-01-28 163842](https://github.com/dune3d/dune3d/assets/9744201/2233feb7-e9a3-4a9a-9a97-d94caef3607f)

Ideally, you should be able to take an item/object/point and select another item/object/point and tell it to align with the second point either vertically, horizontally, or both.

There just appears to be no way to precisely do this without drawing a bunch of constrains then praying your X and Y constraint lines match exactly with the middle.

**Comment by carrotIndustries** [comment]
> however when drawing a reference rectangle there appears to be zero way to align the rectangle to the center of the workspace.


There is: In the draw rectangle tool press <kbd>m</kbd> before placing the first point. This switches it to the center/corner mode.

Alternatively, you can do the same by drawing a diagonal construction line and constraining its center to be on the workplane's origin.

**Comment by biggestsonicfan** [comment]
This seemingly only works if you want a rectangle aligned along two axis, not one?

For example:
![Screenshot 2024-01-29 123434](https://github.com/dune3d/dune3d/assets/9744201/6a18b554-32c6-47f8-8221-7b6767581d03)

While the rectangle is now center with my part, the sketch is now longer than it needs to be on the pinout side. I can delete the construction line and adjust, but there is no way to fine tune the length without using the mouse, which adjusts both the x and y axis, I can't choose a single axis to adjust.

**Comment by carrotIndustries** [comment, marked as answer]
Well, then you have to use a different approach:

![image](https://github.com/dune3d/dune3d/assets/877304/08876c35-f616-4fc2-86e2-f9c044ea33e6)

It's the same thing as in any other parametric CAD.

**Reply by biggestsonicfan** [reply]
Ahh, I see. I guess I'm just too unfamiliar with CADding in general thus far.

Great to see a discussion tab added to the repo, that's much appreciated. I'll see if I can't learn some more basics before coming back with anything I can't figure out.
